packages/react-dom/src/events/SyntheticEvent.js

```diff
@@ -152,11 +152,22 @@ export const UIEventInterface: EventInterfaceType = {
   detail: 0,
 };
 
-let previousScreenX = 0;
-let previousScreenY = 0;
-// Use flags to signal movementX/Y has already been set
-let isMovementXSet = false;
-let isMovementYSet = false;
+let lastMovementX;
+let lastMovementY;
+let lastMouseEvent;
+
+function updateMouseMovementPolyfillState(event) {
+  if (event !== lastMouseEvent) {
+    if (lastMouseEvent && event.type === 'mousemove') {
+      lastMovementX = event.screenX - lastMouseEvent.screenX;
+      lastMovementY = event.screenY - lastMouseEvent.screenY;
+    } else {
+      lastMovementX = 0;
+      lastMovementY = 0;
+    }
+    lastMouseEvent = event;
```

Comment:
Won't this GC root the event? This seems like a leak.

Reply:
It will root only one (the very last) mouse event, and only on IE11, so this should not be an issue.